src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -1057,6 +1194,34 @@ T trace(const SGMatrix<T>& A)
 }
 
 /**
+ * Method that computes the transpose of a matrix.
+ *
+ * @param A The matrix
+ * @return The transposed matrix
+ */
+template <typename T>
+SGMatrix<T> transpose_matrix(const SGMatrix<T>& A)
```

Comment:
Rather than transposing matrices, I think it is a nicer idea to set a flag.
Either this can be added to the data structure (to the new linalg::Matrix, not the SGMatrix), or the linalg methods can have flags that take into account the transpose.

Check out eigen3 `adjoint` vs `transpose`